spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -783,6 +794,682 @@ public void shouldUpdateExistingContentAccessCertContentWhenProductDataChanges()
+        PoolDTO pool = adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        List<CertificateDTO> certs = consumerClient.consumers().fetchCertificates(consumer.getUuid());
+        assertThat(certs).singleElement();
+        String initialCert = certs.get(0).getCert();
+
+        pool.startDate(OffsetDateTime.now().plusDays(1));
+        adminClient.owners().updatePool(ownerKey, pool);
+
+        certs = consumerClient.consumers().fetchCertificates(consumer.getUuid());
+        assertThat(certs).singleElement();
+        String updatedCert = certs.get(0).getCert();
+
+        assertThat(updatedCert).isNotEqualTo(initialCert);
+    }
+
+    @Test
+    public void shouldHaveDisabledPurposeComplianceForOwnerInSCAMode() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        // System purpose status
+        SystemPurposeComplianceStatusDTO status = consumerClient.consumers()
+            .getSystemPurposeComplianceStatus(consumer.getUuid(), null);
+        assertThat(status)
+            .isNotNull()
+            .returns("disabled", SystemPurposeComplianceStatusDTO::getStatus);
+
+        // compliance status
+        ComplianceStatusDTO complianceStatus = consumerClient.consumers()
+            .getComplianceStatus(consumer.getUuid(), null);
+        assertThat(complianceStatus)
+            .isNotNull()
+            .returns("disabled", ComplianceStatusDTO::getStatus);
+    }
+
+    @Test
+    public void shouldRevokeSCACertsUponUnRegistration() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        String ownerKey = owner.getKey();
+
+        ProductDTO prod = adminClient.ownerProducts().createProductByOwner(ownerKey, Products.random());
+        ContentDTO content = adminClient.ownerContent().createContent(ownerKey, Content.random());
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        ConsumerApi consumerApi = new ConsumerApi(consumerClient.getApiClient());
+        Object export = consumerApi.exportCertificates(consumer.getUuid(), null);
+
+        // Impl note: The use of the Request/Reply objects are to avoid a Gson de-serialization
```

Comment:
Unfortunately since the issue is with Gson, we would have to update the Gson object in the generated JSON class. I verified that this will work:
```
JSON json = adminClient.getApiClient().getJSON();
Gson gson = json.getGson();
Gson modified = gson.newBuilder()
    .setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
    .create();
adminClient.getApiClient().getJSON().setGson(modified);

//  Then use the ConsumerApi to export the certs.
```

So, maybe we could do this in the ConsumerClient class for the fetchCertificates method, and then set the Gson object back to the originally generated one in a finally block? That would allow us to just use the fetchCertificates method and get the correct serial value.

Reply:
Yeah. We should fix it in the client to avoid repeating this workaround around. The fix doesn't seem to be breaking anything else, so we could probably just add it with a note to remove it once it gets fixed. Is there a card that tracks the issue on the OpenAPI's side?

Reply:
Looking at this again, we really should update the object to number strategy for **ALL** clients that are created. I think that we should update this value in the constructor of org.candlepin.spec.bootstrap.client.ApiClient. This will resolve the issue for all endpoints that run into this truncation issue.  I updated the code to take this approach!

Reply:
After making these changes, I noticed that the tests that compare serial values would occasionally fail for the same issue even though we updated the Gson object to use `ToNumberPolicy.BIG_DECIMAL` as the object to number strategy for all ApiClients that are constructed.

The issue is that there is an issue with parallel execution because whenever a org.candlepin.invoker.client.ApiClient is created in the ApiClientFactory, the static JSON object (which contains the static Gson object) is recreated with the default serialization configurations. So, there is a brief time where this Gson object is references by another test and the certificate serial value is de-serialized as a double and is then inaccurate.

I almost think we have to use our Request/Response objects for making this request at this point.

Reply:
You are right. With the static linking OpenApi uses to setup GSON, there is no way around that. Request looks like the only way forward. Regarding the card. If there is none we should create one so that they can fix it.

We should also revisit tests that are changing gson (there are some that add serializeNulls) to avoid this issue there as well.

Reply:
I updated the PR to use our Request class in the ConsumerClient. I also opened an issue for openapi-generator and logged a jira for updating the tests that change the Gson object.

https://github.com/OpenAPITools/openapi-generator/issues/14372

https://issues.redhat.com/browse/CANDLEPIN-452